### PR TITLE
feat(keep): Google Keep bridge v1 (issue #1000)

### PR DESCRIPTION
**Pull request by brendanjerwin**
## Summary

- First slice of the Google Keep bridge from #1000 — connect/bind UX, full Keep wire-protocol port to Go, gRPC service, help docs.
- Ships **verification-only sync** in v1 (the wiki periodically confirms the master token still works and the bound note still exists). Bidirectional data sync is tracked in #1007.
- Honest about the trust model in `help_google_keep`: this is an unofficial reverse-engineered API, master tokens are credential-equivalent, may break when Google changes the wire format.

## What works end-to-end after this lands

1. Visit `/profile` → see the **Google Keep** section.
2. Generate a Google App-Specific Password (2-Step Verification required).
3. Enter email + ASP, click **Test & Save** → wiki performs gpsoauth exchange, verifies against Keep, stores the master token (ASP is discarded).
4. Open any page with a `{{ Checklist "name" }}` macro → see a **🔗 Bind to Keep List** button.
5. Click bind → picker shows your existing Keep list-typed notes or offers to create a new one.
6. After bind: status pill replaces the button; the bound note appears in your Keep app on your phone.
7. Disconnect / unbind / rebind all work; multi-user (per-Tailscale-identity) bindings each live on their own profile so Alice and Bob can each bind the same wiki list to their own Keep notes.

What does **not** round-trip in v1: actual data sync. That's #1007.

## What's in the diff

| Layer | Files | Notes |
|---|---|---|
| Wire protocol port | `internal/keep/protocol/` | gpsoauth Stage 1+2, typed `/notes/v1/changes` decoder, CreateList push, REFERENCE.md pinning the gkeepapi+gpsoauth commits we mirrored. ~80 Ginkgo specs. |
| Bridge layer | `internal/keep/bridge/` | Bindings storage with per-user collision matrix, Connector orchestrator, wiki↔Keep field mapping with #tag round-trip. |
| Proto + gRPC | `api/proto/api/v1/keep_connector.proto`, `internal/grpc/api/v1/keep_connector.go` | 8 RPCs scoped to calling user via Tailscale identity. |
| UI | `static/js/web-components/keep-connect.ts`, `keep-bind-button.ts`, edits to `wiki-checklist.ts` | Macro on profile page; bind button rendered inside the existing Checklist component (only visible when calling user has KeepConnect configured). |
| Help | `internal/syspage/embedded/help_google_keep.md` (new) + edits to `help.md`, `help_macro_checklist.md`, `help_profile.md` | Trust model, ASP setup, multi-user binding semantics, error matrix, "for agents" section. |

## TDD spine

Per CLAUDE.md, every new capability landed as Skeleton → Red → Green → Refactor commits within the larger feature commits. The wire-protocol layer has full unit-test coverage. The bridge bindings have a 26-spec collision matrix. Mapping has table-driven round-trip tests. Lit components ship without unit tests in this PR (tracked in #1007 alongside the real sync work).

## Plan-vs-code review (self-audit)

Ran the mandatory plan-vs-code reviewer subagent. Findings:
- ✅ Master token storage matches plan (profile-page frontmatter, plaintext, Tailscale-fronted).
- ✅ All RPCs scope to calling user; no cross-user leak path.
- ✅ ASP not retained after exchange (verified at code level).
- ✅ Bind UI only shows when calling user has KeepConnect configured.
- ⚠️ Sync engine (reconcile + tick + scheduler + echo suppression) is the explicit v1 scope reduction → tracked in #1007 and disclosed in `help_google_keep.md`.
- ⚠️ Lit component tests deferred → also #1007.

## Test plan

- [ ] CI green (Go tests, FE tests, lint:everything).
- [ ] `devbox services start` locally — visit `/profile`, see KeepConnect macro render.
- [ ] Connect with a real Google account (test account, NOT primary) — observe master token written, ASP discarded.
- [ ] Open a checklist page → see Bind button.
- [ ] Bind to a new Keep note → verify it appears in the Keep app.
- [ ] Disconnect → bindings show paused; reconnect → resume.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by brendanjerwin** [comment]
@claude — coverage gate is failing on PR #1008. After excluding cmd/keep-debug (a diagnostic CLI), projected coverage is ~76%, still 4-5% under SonarCloud's 80% threshold on new code.

Top remaining uncovered files (Go + TS):
- `internal/grpc/api/v1/keep_connector.go` — 144/178 lines uncovered. Existing test file covers ListDeadLetters/ClearDeadLetter only; missing handlers: ExchangeAndStore, Disconnect, GetState, ListMyBindings, ListNotes, BindChecklist, UnbindChecklist, GetChecklistBindingState.
- `internal/keep/bridge/sync_debouncer.go` — 49/49 (no test file).
- `internal/keep/bridge/cron.go` — 47/56 (existing tests don't cover the registry/discovery walk).
- `internal/bootstrap/server.go` — 86/87 from the new `setupGRPCServer` body.
- `static/js/web-components/keep-connect.ts` — 58/58 (no test file).
- `static/js/web-components/keep-bind-button.ts` — 54/54 (no test file).

Please add unit tests bringing new-code coverage above 80%. Match existing patterns:
- gRPC handlers: see `internal/grpc/api/v1/keep_connector_test.go` for the fakeStore/fakeAuth/fakeKeepClient harness pattern.
- Bridge components: see `internal/keep/bridge/*_test.go` for ginkgo + freshConnector patterns.
- Lit components: see `static/js/web-components/*.test.ts` for fixture/sinon patterns; mock the gRPC client per `feedback_*` patterns in CLAUDE.md.

Don't introduce new tests against fragile UI internals — test public behavior, state transitions, and event wiring.

**Comment by brendanjerwin** [comment]
@claude — coverage now at **76.1%** (up from 42.3%). Still 4% short of the 80% gate. Top remaining gaps:

- `internal/keep/bridge/connector.go` — **315/1243 uncovered**. Existing tests hit the divergence rule's main rows; gaps are likely in `MigrateBindingFingerprints` paths, `Bind` flow edge cases, label CRUD branches, and the various error-return arms in SyncToKeep/applyInboundFromKeep.
- `internal/keep/protocol/client.go` — **107/329 uncovered**. Likely in CreateListWithItems error paths and decoder edge cases (writeResults shapes).
- `server/checklistmutator/sync_helpers.go` — **55/58 uncovered**. Just needs unit tests for AddItemForSync / UpdateItemForSync / DeleteItemForSync / parseSortHint.
- `internal/keep/protocol/gpsoauth.go` — **29/94 uncovered**. classifyAuthError branches, ExchangeOAuthTokenForMasterToken error paths.
- `static/js/eslint-rules/no-string-error-on-litelement.js` — **29/29 uncovered**. Add a small ESLint rule unit test using `RuleTester`.

Skip these (already excluded or near-zero ROI):
- `internal/bootstrap/server.go` (integration glue, intentionally untested at unit level)

Push above 80% to clear the gate.

File: internal/keep/bridge/bindings.go [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+
+// Binding is a single user's link from a wiki checklist (page + list_name)
+// to a Keep note in their account.
+type Binding struct {
+	Page          string
+	ListName      string
+	KeepNoteID    string
+	KeepNoteTitle string
+	BoundAt       time.Time
+}
```
[the remark this answers is not in the document]
Comment (brendanjerwin) [reply internal/keep/bridge/bindings.go:144]: The proto BindingState exposes those fields for surfacing to the UI, but the Go struct intentionally splits them: stable per-user config (page, list_name, keep_note_id, keep_note_title, bound_at) lives in `wiki.connectors.google_keep.bindings[]`, while the high-churn sync metadata (cursor, last_pull_at, last_push_at, last_verified_at, last_error) lives in the parallel `agent.connectors.google_keep.bindings[<keep_note_id>].*` namespace per the original plan. The gRPC handler joins them at query time. Some of those metadata fields land in v2 (#1007); for v1 the gRPC returns the BindingState shape with the available fields populated and the rest zeroed.

File: internal/keep/bridge/bindings.go [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+// lockProfile acquires the per-profile mutex.
+func (s *BindingStore) lockProfile(profileID wikipage.PageIdentifier) func() {
+	v, _ := s.profilMu.LoadOrStore(profileID, &sync.Mutex{})
+	mu, ok := v.(*sync.Mutex)
+	if !ok {
+		mu = &sync.Mutex{}
+	}
+	mu.Lock()
+	return mu.Unlock
+}
```
[the remark this answers is not in the document]
Comment (brendanjerwin) [reply internal/keep/bridge/bindings.go:388]: The current implementation already does what you suggested: it does a checked type assertion on `*sync.Mutex` and panics with a clear message if the assertion fails (lines 382-385 in the latest revision). The local-mutex fallback you flagged was an earlier draft. On the sync.Map memory-growth concern: this is per-profile (per-user), and the wiki is household-scale; the map is bounded by the number of users who ever connect Keep, which is single-digit in practice. Pruning isn't worth the additional complexity here.
